src/lib/game/world/content-queue.js

```diff
@@ -0,0 +1,83 @@
+class ContentQueue {
+
+  constructor(processor, interval = 1, workFactor = 1, minWork = 1) {
+    this.processor = processor;
+
+    this.interval = interval;
+    this.workFactor = workFactor;
+    this.minWork = minWork;
+
+    this.queue = new Map();
+
+    this.schedule = ::this.schedule;
+    this.run = ::this.run;
+
+    this.schedule();
+  }
+
+  has(keyPattern) {
+    // todo: allow wildcards (ie 8-*)
+    return this.queue.has(keyPattern);
+  }
+
+  add(key, job) {
+    if (this.queue.has(key)) {
+      return;
+    }
+
+    this.queue.set(key, job);
+  }
+
+  remove(keyPattern) {
+    let count = 0;
+
+    if (keyPattern.toString().includes('*')) {
+      for (const entry of this.queue) {
+        const key = entry[0];
+
+        if (key.startsWith(keyPattern.split('*')[0])) {
+          this.queue.delete(key);
+          count++;
+        }
```

Comment:
Ah, I see! :+1: If not, it would probably be good to remove it for now (as well as the TODO a bit above it). Great idea to add at some point though. I feel it could be an extension to Map maybe, then we could have the queue itself handle the matching in `has` and `remove` for example.